cmd/controller/main.go

```diff
@@ -213,14 +214,14 @@ func main() {
 		ctlConf.MinPort, ctlConf.MaxPort, ctlConf.SidecarImage, ctlConf.AlwaysPullSidecar,
 		ctlConf.SidecarCPURequest, ctlConf.SidecarCPULimit,
 		ctlConf.SidecarMemoryRequest, ctlConf.SidecarMemoryLimit, ctlConf.SdkServiceAccount,
-		kubeClient, kubeInformerFactory, extClient, agonesClient, agonesInformerFactory)
+		kubeClient, kubeInformerFactory, extClient, agonesClient, agonesInformerFactory, ctlConf.SplitController)
```

Comment:
Rather than burying the if (!split) { NewExtensions() } in the NewController call for each of these, I think it's clearer if you instead just lift the NewExtensions() call up to here (main.go).

I personally would just keep the block of NewController calls here, then around like 223 after all the controllers are initialized, have a block like:

```
if (!<feature flag is enabled>) {
  foo.NewExtensions()
  ...
}
```

Later when we go fold the feature flag to "true", we'd just delete that whole block, which makes that diff less messy.

Reply:
Thanks for pointing this out. I should've realized this since it is similar what it's doing in extensions. 😅